### PR TITLE
AV-175659 : ingestion layer changes for SSORule CRD

### PR DESCRIPTION
**Pull request by DixitAakash**

### CODE REVIEW

File: internal/k8s/validator.go [hunk 1]
```diff
@@ -371,6 +391,113 @@ func (l *leader) ValidateServiceImportObj(key string, serviceImport *akov1alpha1
...
+					clientSecretString := string(clientSecretObj.Data["clientSecret"])
+					if clientSecretString == "" {
+						err = fmt.Errorf("clientSecret field not found in %s secret", clientSecret)
+						status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+				}
+
+				if profile.ResourceServer != nil {
+					if *profile.ResourceServer.AccessType == "ACCESS_TOKEN_TYPE_JWT" && profile.ResourceServer.JwtParams == nil {
```
Comment (swathinsankaran) [inline internal/k8s/validator.go:444]: a nil check seems to be missing for the AccessType or is it having a default value?

File: internal/k8s/validator.go [hunk 2]
```diff
@@ -371,6 +391,113 @@ func (l *leader) ValidateServiceImportObj(key string, serviceImport *akov1alpha1
...
+						return err
+					}
+					if *profile.ResourceServer.AccessType == "ACCESS_TOKEN_TYPE_OPAQUE" && profile.ResourceServer.OpaqueTokenParams == nil {
+						err = fmt.Errorf("Access Type is %s, but Opaque Token Params have not been specified", *profile.ResourceServer.AccessType)
+						status.UpdateOAuthSamlConfigStatus(key, oauthSamlConfig, status.UpdateCRDStatusOptions{Status: lib.StatusRejected, Error: err.Error()})
+						return err
+					}
+
+					if profile.ResourceServer.OpaqueTokenParams != nil {
+						serverSecret := *profile.ResourceServer.OpaqueTokenParams.ServerSecret
```
Comment (swathinsankaran) [inline internal/k8s/validator.go:456]: Do we need a nil check on ServerSecret?

File: internal/lib/control_config.go [hunk 3]
```diff
@@ -244,6 +263,18 @@ func (c *akoControlConfig) SetCRDEnabledParams(cs akocrd.Interface) {
...
 
+func (c *akoControlConfig) Setv1alpha2CRDEnabledParams(cs v1alpha2akocrd.Interface) {
+	timeout := int64(120)
+	_, oauthSamlErr := cs.AkoV1alpha2().OAuthSamlConfigs(metav1.NamespaceAll).List(context.TODO(), metav1.ListOptions{TimeoutSeconds: &timeout})
+	if oauthSamlErr != nil {
+		utils.AviLog.Infof("ako.vmware.com/v1alpha2/OAuthSamlConfig not found/enabled on cluster: %v", oauthSamlErr)
+		c.oauthSamlConfigEnabled = false
+	} else {
+		utils.AviLog.Infof("ako.vmware.com/v1alpha2/OAuthSamlConfig enabled on cluster")
+		c.oauthSamlConfigEnabled = true
```
Comment (swathinsankaran) [inline internal/lib/control_config.go:274]: We have to honor Akshay's new changes related to the CRD enable flag once his changes are merged into the master.

File: internal/k8s/validator.go (hunk 1 above)
Comment (DixitAakash) [reply internal/k8s/validator.go:444]: AccessType is a required field in the CRD, so nil check is not required.

File: internal/k8s/validator.go (hunk 2 above)
Comment (DixitAakash) [reply internal/k8s/validator.go:456]: ServerSecret is a required field in the CRD, so nil check is not required.

File: internal/lib/control_config.go (hunk 3 above)
Comment (DixitAakash) [reply internal/lib/control_config.go:274]: Sure, I have noted this.

**Pull request merged**